instrumentation/aws-sdk/src/main/java/brave/instrumentation/aws/TracingRequestHandler.java

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright 2016-2018 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package brave.instrumentation.aws;
+
+import brave.Span;
+import brave.Tracing;
+import com.amazonaws.AmazonServiceException;
+import com.amazonaws.AmazonWebServiceRequest;
+import com.amazonaws.AmazonWebServiceResult;
+import com.amazonaws.Request;
+import com.amazonaws.Response;
+import com.amazonaws.ResponseMetadata;
+import com.amazonaws.handlers.HandlerAfterAttemptContext;
+import com.amazonaws.handlers.HandlerContextKey;
+import com.amazonaws.handlers.RequestHandler2;
+import zipkin2.Endpoint;
+
+public class TracingRequestHandler extends RequestHandler2 {
```

Comment:
I'd suggest making a normal `TracingRequestHandler` possibly based on normal http clients, but it could be a non-http "application span" as well (considering underneath might be traced as well). At any rate make a factory method and builder for normal use cases. Hide the ctor package private as usual.

For service loader, make a delegate similar to our DelegatingTracingFilter, maybe named CurrentTracingRequestHandler and in that override a method like `Tracer tracer()` using `Tracing.currentTracer()`. That way the explicitly configured one acts normally and via a delegate a service-loader instantiated one uses the implicit stuff. make sense?

Reply:
Ok, actually, looking through their examples, it looks like you can add it at client create time. So we cover both cases, the user specifies one, and the one that gets injected automatically with the `request.handler2s` file